examples/all-clusters-app/all-clusters-common/src/tls-client-management-instance.cpp

```diff
@@ -0,0 +1,171 @@
+/*
+ *
+ *    Copyright (c) 2025 Matter Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/clusters/tls-certificate-management-server/CertificateTableImpl.h>
+#include <app/clusters/tls-client-management-server/tls-client-management-server.h>
+#include <clusters/TlsClientManagement/Commands.h>
+#include <tls-client-management-instance.h>
+#include <vector>
+
+static constexpr uint8_t kMaxProvisioned = 254;
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::DataModel;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::Tls;
+using namespace chip::app::Clusters::TlsClientManagement;
+using namespace Protocols::InteractionModel;
+
+CHIP_ERROR TlsClientManagementCommandDelegate::GetProvisionedEndpointByIndex(EndpointId matterEndpoint, FabricIndex fabric,
+                                                                             size_t index, EndpointStructType & endpoint) const
+{
+    VerifyOrReturnError(matterEndpoint == EndpointId(1), CHIP_ERROR_INTERNAL);
+
+    size_t fabricI = 0;
+    for (auto i = mProvisioned.begin(); i != mProvisioned.end(); i++)
+    {
+        if (i->fabric == fabric)
+        {
+            if (fabricI++ == index)
+            {
+                endpoint = i->payload;
+                return CHIP_NO_ERROR;
+            }
+        }
+    }
+
+    return CHIP_ERROR_PROVIDER_LIST_EXHAUSTED;
+}
+
+Status TlsClientManagementCommandDelegate::ProvisionEndpoint(
+    EndpointId matterEndpoint, FabricIndex fabric,
+    const TlsClientManagement::Commands::ProvisionEndpoint::DecodableType & provisionReq, uint16_t & endpointID)
+{
+    VerifyOrReturnError(matterEndpoint == EndpointId(1), Status::ConstraintError);
+
+    if (mCertificateTable.HasRootCertificateEntry(fabric, provisionReq.caid) != CHIP_NO_ERROR)
+    {
+        return Status::ConstraintError;
+    }
+    if (!provisionReq.ccdid.IsNull() &&
+        mCertificateTable.HasClientCertificateEntry(fabric, provisionReq.ccdid.Value()) != CHIP_NO_ERROR)
+    {
```

Comment:
Per Spec, this should be `ClientCertNotFound` status code.

Reply:
Fixed